lib/logging.dart

```diff
@@ -88,29 +88,29 @@ class Logger {
   Logger._internal(this.name, this.parent, Map<String, Logger> children)
       : this._children = children,
         this.children = new UnmodifiableMapView(children) {
-    if (parent != null) parent._children[name] = this;
+    /* if (parent != null) */ parent._children[name] = this;
```

Comment:
I expected `parent` would be nullable, I'd guess it must be null on the root logger.

Can you dig on why this choice was made?

Reply:
See my other comment on this line (and on line 67).  Short story: it's wrong; it's happening because of https://github.com/dart-lang/sdk/issues/38341.